ocaml/xapi/record_util.ml

```diff
@@ -198,6 +198,25 @@ let string_to_network_default_locking_mode = function
   | "disabled" -> `disabled
   | s -> raise (Record_failure ("Expected 'unlocked' or 'disabled', got "^s))
 
+let network_purpose_to_string = function
+  | `nbd -> "NBD"
+  | `insecure_nbd -> "Insecure_NBD"
+  | `management -> "Management"
+  | `storage -> "Storage"
+  | `guest -> "Guest"
+  | `himn -> "Host_Internal_Management"
+  | `unmanaged -> "Unmanaged"
+
+let string_to_network_purpose = function
```

Comment:
Since these are the params the user will have to give on the CLI, I think we should make these lowercase. It seems that for most, or maybe all of the other CLI parameters are lowercase. Maybe even nbd should be lowercase, not sure about that. Also, it seems to me that most CLI params have hyphens in them, not underscores, and that's probably also easier to type. Although I did find one param type that contains `_`s for some reason, `Record_util.host_display_to_string`.

Reply:
The `..._to_string` functions are used for logging, and the `string_to_...` functions are used for reading input from users, and there's a bit of tension between the ideal strings in the two cases.

Also, some of these purposes should never be set (or unset) by users, e.g. HIMN which is created and configured automatically.

Still, I don't object to having them lowercase and hyphenated.